ThomasAwesome shuffle mobbers

dignite opened the pull request:
This is an attempt to revive #18 which is a great feature.

**As** a user of the app
**In order** to mix things up
**And** not having the same person that I am taking over from
**And** not having the same person that I am handing over to
**And** not having to remove and re-add mobbers to achieve this
**I want** to be able to shuffle the order of the mobbers

![image](https://user-images.githubusercontent.com/2315367/49178134-4297a300-f34f-11e8-9bc4-231103657434.png)

allan-stewart commented on src/windows/config/index.html:
```diff
@@ -15,6 +15,9 @@
       </div>
       <div class="mobbers">
         <h1>Mobbers</h1>
+        <div class="mobberActions">
```
Hmm... isn't adding a mobber also a "mobber action?" Not sure if this is the right name given the placement, but it's a minor quibble.

dignite replied:
Good point! I'm open to suggestions on a better name, will think about that one

allan-stewart commented:
The placement of the button is kind of bugging me, and I'm trying to decide if it's worth changing.

In the current placement, the shuffle button looks like it's the first mobber on the list (especially with the gray striping). If you have mobbers already, then maybe that's where you want the button. But if there aren't any mobbers yet then does it distract from adding them?

Maybe the button should go at the bottom, under the "Add" form? If you have a bunch of mobbers it would push the button down so you have to scroll though. But overall, I'm thinking I like it under better than on top... thoughts?

dignite commented:
I tried both and settled on above, but I am okay either way. One idea could be to keep it where it is but disable/hide the shuffle if there are no mobbers

allan-stewart commented:
After trying it a few ways, I feel like the top placement improves without the gray background. The striping feels like it's part of the list of mobbers. But when it's white, it blends back into the page background and the first mobber has the gray background.

Only showing it for 2+ mobbers is an interesting idea. It gets it out of the way when you can't really use it and stops it from distracting new users from adding their mobbers. But then it may not be obvious why it's appearing / disappearing. Disabling for < 2 mobbers is a nicer touch, but doesn't solve the placement problem. /shrug

I'll try to get some other people to voice their opinions on the placement.

ChuckBates commented:
I definitely don't like the gray striping, but I'm not a fan of the while either as there's no separation from this button's row and the mobber list. Is there another color that fits the UX theme that we can use to help distinguish this as distinct from the mobber list but also not bland?

dignite commented:
- Moved shuffle button below "add mobber" form
- Renamed class `mobberActions` to `mobberListActions` to signal actions on the list as a whole
- Attempted writing a test in `timer-state.spec.js`, mobbers dependency is not easily stubable with the current code. Not writing a test at this time but I think testing with a mocked `mobbers` is a good idea once that is possible.

![image](https://user-images.githubusercontent.com/2315367/49178272-a15d1c80-f34f-11e8-9dc5-9805892b562e.png)

allan-stewart commented:
Fair enough re: the test. I'll take a stab at it post-merge.
